### PR TITLE
Macos integration

### PR DESCRIPTION
**Pull request by olegnet**
(Copy of `macos/README.md`)

1. Run `flutter create --platforms=macos .`
2. Remove unnecessary new files created in step 2
3. Add some lines from `ios/mylib.podspec` to `macos/mylib.podspec`
4. Replace `macos/Classes/*` with `ios/Classes/*`
5. Few more changes in source code to compatibility with new ffi and ffigen

If you start `flutter run` at this point you will get something like that
```
══╡ EXCEPTION CAUGHT BY WIDGETS LIBRARY ╞═══════════════════════════════════════════════════════════
The following ArgumentError was thrown attaching to the render tree:
Invalid argument(s): Failed to lookup symbol (dlsym(RTLD_DEFAULT, rust_greeting): symbol not found)
```

You can try changing `example/macos/Podfile` to resolve this error, but there is [one more recipe](https://flutter.dev/docs/development/platform-integration/c-interop#compiled-dynamic-library-macos) to do that.

### CODE REVIEW

File: rust/makefile [hunk 1]
```diff
@@ -52,6 +53,16 @@ target/universal/release/libexample.a: $(SOURCES) ndk-home
 	fi
 	@echo "[DONE] $@"
 
+## macos: Compile the macos libraries
+macos: target/x86_64-apple-darwin/release/libexample.dylib
+
+target/x86_64-apple-darwin/release/libexample.dylib: $(SOURCES) ndk-home
```
Comment (brickpop) [inline rust/makefile:59]: Why only adding a target for x86_64 and not `aarch64`?
Why does the target depend on ndk-home (Android)?

File: rust/makefile [hunk 2]
```diff
@@ -52,6 +53,16 @@ target/universal/release/libexample.a: $(SOURCES) ndk-home
...
+	@if [ $$(uname) == "Darwin" ] ; then \
+		cargo lipo --release --targets x86_64-apple-darwin ; \
+		else echo "Skipping iOS compilation on $$(uname)" ; \
```
Comment (brickpop) [inline rust/makefile:62]:
```suggestion
		else echo "Skipping macOS compilation on $$(uname)" ; \
```

**Review by brickpop — COMMENTED** [review]
Hi @olegnet 
Thank you for the effort of adding MacOS as a target.
I have almost no time to check and review the changes myself, but so far I have a couple of minor comments (see below)

**Comment by brickpop** [comment]
Besides that, could you confirm if generating release builds of a package consumer app works properly for you?
On iOS, the linker will strip the C symbols that are not explicitly invoked from real swift code. There is an issue with a few things to try, but no time to work on that yet (https://github.com/brickpop/flutter-rust-ffi/issues/16)

File: rust/makefile (hunk 2 above)
Comment (olegnet) [reply rust/makefile]: fixed

**Comment by olegnet** [comment]
I removed changes in Intellij Idea configuration files from "part 2" commit and did final changes in "part 3".
There's definitely some more work with release builds but as far as I checked it works locally. 
It's different from iOS case because we have dynamic library like with Android

**Pull request merged**